### PR TITLE
patches: add gst-libav avvideocompare

### PR DESCRIPTION
**Pull request by uartie**
Signed-off-by: U. Artie Eoff <ullysses.a.eoff@intel.com>

### CODE REVIEW

**Comment by uartie** [comment]
https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/3366

**Comment by HeJunyan** [comment]
Yeah, good work. I noticed that before. It seems that we can use it do compare our encoder quality in CI easily.

It seems that the ffmpeg has included ssim and psnr algorithm and a a X86 optimized version, which is lack in gstreamer. So we need to depends on that.

File: patches/0050-gst-libav-add-avvideocompare-element.patch [hunk 1]
```diff
@@ -0,0 +1,801 @@
...
++    GST_OBJECT_UNLOCK (self);
++    caps1 = gst_pad_get_current_caps (self->sinkpad1);
++    caps2 = gst_pad_get_current_caps (self->sinkpad2);
++    GST_ELEMENT_ERROR (self, CORE, NEGOTIATION, (NULL),
++        ("input formats don't match: %" GST_PTR_FORMAT " vs. %" GST_PTR_FORMAT,
++            caps1, caps2));
++    if (caps1)
++      gst_caps_unref (caps1);
++    if (caps2)
++      gst_caps_unref (caps2);
```
Comment (xhaihao) [inline patches/0050-gst-libav-add-avvideocompare-element.patch:766]: You may call `gst_caps_unref()` without the check against NULL.

File: patches/0050-gst-libav-add-avvideocompare-element.patch [hunk 2]
```diff
@@ -0,0 +1,801 @@
...
++    return GST_FLOW_ERROR;
++  }
++map_failed:
++  {
++    GST_OBJECT_UNLOCK (self);
++    GST_DEBUG_OBJECT (self, "Failed to map frame");
++    if (buf2)
++      gst_buffer_unref (buf2);
++    if (buf1)
++      gst_buffer_unref (buf1);
```
Comment (xhaihao) [inline patches/0050-gst-libav-add-avvideocompare-element.patch:776]: Here the check is unnecessary too.

File: patches/0050-gst-libav-add-avvideocompare-element.patch (hunk 1 above)
Comment (uartie) [reply patches/0050-gst-libav-add-avvideocompare-element.patch]: updated

File: patches/0050-gst-libav-add-avvideocompare-element.patch (hunk 2 above)
Comment (uartie) [reply patches/0050-gst-libav-add-avvideocompare-element.patch]: updated

**Pull request merged**